homeassistant/components/cloudflare/__init__.py

```diff
@@ -97,20 +93,50 @@ async def _async_update_cloudflare(
 ) -> None:
     _LOGGER.debug("Starting update for zone %s", dns_zone["name"])
 
-    records = await client.list_dns_records(zone_id=dns_zone["id"], type="A")
+    records = await client.list_dns_records(zone_id=dns_zone["id"])
     _LOGGER.debug("Records: %s", records)
 
     session = async_get_clientsession(hass, family=socket.AF_INET)
-    location_info = await async_detect_location_info(session)
-
-    if not location_info or not is_ipv4_address(location_info.ip):
-        raise HomeAssistantError("Could not get external IPv4 address")
+    session_ipv6 = async_get_clientsession(hass, family=socket.AF_INET6)
 
-    filtered_records = [
-        record
-        for record in records
-        if record["name"] in target_records and record["content"] != location_info.ip
-    ]
+    location_info = await async_detect_location_info(session)
+    location_info_v6 = await async_detect_location_info(session_ipv6)
+
+    filtered_records = []
+
+    if location_info and is_ipv4_address(location_info.ip):
+        _LOGGER.info("IPv4 address detected: %s", location_info.ip)
+        for record in records:
+            if (
+                record["name"] in target_records
+                and record["content"] != location_info.ip
+                and record["type"] == "A"
+            ):
+                _LOGGER.info(
+                    "Will update record %s from %s to %s",
```

Comment:
This should probably remain `info` level although perhaps more verbosity may be helpful.
```suggestion
                    "IPv4 address change detected for record: '%s', will DNS entry from %s to %s",
```

Reply:
Although arguably, it may be better to switch these to being debug levels and then have a `info` level log after the update(s) is done
https://github.com/home-assistant/core/blob/da3ba5f8a350130609437f2e58d4a8a1ce950f43/homeassistant/components/cloudflare/__init__.py#L159
Another option would be to leave these as info and then log an error if the update fails.